src/models/dataStore.ts

```diff
@@ -76,7 +79,11 @@ export async function saveNewInstance(d2: D2, instance: any): Promise<Response>
 export async function deleteInstance(d2: D2, instance: any): Promise<Response> {
     try {
         const instanceArray = await getDataStore(d2, instancesKey);
-        const newInstanceArray = _.differenceWith(instanceArray, [instance], _.isEqual);
+        const newInstanceArray = _.differenceWith(
+            instanceArray,
+            [instance],
+            (inst: InstanceData) => inst.id === instance.id
+        );
```

Comment:
Proposal:

```
        const instanceArray: InstanceData[] = await getDataStore(d2, instancesKey);
        const newInstanceArray = instanceArray.filter(inst => inst.id !== instance.id);
```

Reply:
I'll change this logic in PR #31 where the datastore gets altered

Reply:
done